demos/browser/app/meetingV2/meetingV2.ts

```diff
@@ -1169,15 +1191,112 @@ export class DemoMeetingApp
+      this.videoDownstreamMetricsKeyStatsToShow,
+      videoUpstreamMetricReport[attendeeId]
+    );
+  }
+
+  showVideoStats = (
+    tileIndex: number,
+    streamDirection: StreamDirection,
+    keyStatstoShow: { [key: string]: string },
+    metricsData: { [id: string]: {[key: string]: number} }
+  ): void => {
+    const streams = metricsData ? Object.keys(metricsData) : [];
+    if (streams.length === 0) {
+      return;
+    }
+
+    let statsInfo: HTMLDivElement = document.getElementById(
+      `stats-info-${tileIndex}`
+    ) as HTMLDivElement;
+    if (!statsInfo) {
+      statsInfo = document.createElement('div');
+      statsInfo.setAttribute('id', `stats-info-${tileIndex}`);
+      statsInfo.setAttribute('class', `stats-info`);
+    }
+
+    const statsInfoTableId = `stats-table-${tileIndex}`;
+    let statsInfoTable = document.getElementById(statsInfoTableId) as HTMLTableElement;
+    if (statsInfoTable) {
+      statsInfo.removeChild(statsInfoTable);
+    }
+    statsInfoTable = document.createElement('table') as HTMLTableElement;
+    statsInfoTable.setAttribute('id', statsInfoTableId);
+    statsInfoTable.setAttribute('class', 'stats-table');
+    statsInfo.appendChild(statsInfoTable);
+
+    const videoEl = document.getElementById(`video-${tileIndex}`) as HTMLVideoElement;
+    videoEl.insertAdjacentElement('afterend', statsInfo);
+    const header = statsInfoTable.insertRow(-1);
+    let cell = header.insertCell(-1);
+    cell.innerHTML = 'Video Metrics';
+    for (let cnt = 0; cnt < streams.length; cnt++) {
+      cell = header.insertCell(-1);
+      cell.innerHTML = `${streamDirection} ${cnt + 1}`;
+    }
+
+    for (const [key, value] of Object.entries(keyStatstoShow)) {
+      const row = statsInfoTable.insertRow(-1);
+      row.setAttribute('id', `${streamDirection}-${key}-${tileIndex}`);
+      cell = row.insertCell(-1);
+      cell.innerHTML = value;
+    }
+
+    for (const ssrc of streams) {
+      let frameHeight, frameWidth;
+      if (streamDirection === StreamDirection.Downstream) {
+        frameHeight = metricsData[ssrc]['videoDownstreamGoogFrameHeight'] ? metricsData[ssrc]['videoDownstreamGoogFrameHeight'] : metricsData[ssrc]['videoDownstreamFrameHeight'];
+        frameWidth = metricsData[ssrc]['videoDownstreamGoogFrameWidth'] ? metricsData[ssrc]['videoDownstreamGoogFrameWidth'] : metricsData[ssrc]['videoDownstreamFrameWidth'];
+      } else if (streamDirection === StreamDirection.Upstream) {
+        frameHeight = metricsData[ssrc]['videoUpstreamGoogFrameHeight'] ? metricsData[ssrc]['videoUpstreamGoogFrameHeight'] : metricsData[ssrc]['videoUpstreamFrameHeight'];
+        frameWidth = metricsData[ssrc]['videoUpstreamGoogFrameWidth'] ? metricsData[ssrc]['videoUpstreamGoogFrameWidth'] : metricsData[ssrc]['videoUpstreamFrameWidth'];
+      }
+      if (frameHeight && frameWidth) {
+        const row = document.getElementById(
+          `${streamDirection}-resolution-${tileIndex}`
```

Comment:
Can we add a row header to show streamDirection? Instead of showing it in every row like this? It makes it a bit too long and may be bad in smaller screen.
Also have you verified how it looked in mobile?